src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Services/UserProfileService.cs

```diff
@@ -35,13 +35,40 @@ public UserProfileService(HttpClient httpClient, IOptions<GeneralSettings> setti
         }
 
         /// <inheritdoc/>
-        public async Task<UserProfile> GetUser(string ssn)
+        public async Task<UserProfile> GetUser(string ssnOrExternalIdentity)
         {
             UserProfile user = null;
             DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(UserProfile));
 
             Uri endpointUrl = new Uri($"{_settings.BridgeProfileApiEndpoint}users");
-            StringContent requestBody = new StringContent(JsonSerializer.Serialize(ssn), Encoding.UTF8, "application/json");
+            StringContent requestBody = new StringContent(JsonSerializer.Serialize(ssnOrExternalIdentity), Encoding.UTF8, "application/json");
+
+            HttpResponseMessage response = await _client.PostAsync(endpointUrl, requestBody);
+            if (response.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                Stream stream = await response.Content.ReadAsStreamAsync();
+                user = serializer.ReadObject(stream) as UserProfile;
+            }
+            else
+            {
+                _logger.LogError($"Creating user failed {response.StatusCode}");
```

Comment:
Logging statement in GetUser will log that "Creating user failed" and CreateUser will log "Getting user failed"